app/src/views/private/components/image-editor.vue

```diff
@@ -3,19 +3,18 @@ import api, { addTokenToURL } from '@/api';
 import { useSettingsStore } from '@/stores/settings';
 import { getRootPath } from '@/utils/get-root-path';
 import { unexpectedError } from '@/utils/unexpected-error';
+import type { File } from '@directus/types';
 import Cropper from 'cropperjs';
+import { isEqual } from 'lodash';
 import throttle from 'lodash/throttle';
 import { nanoid } from 'nanoid/non-secure';
 import { computed, nextTick, reactive, ref, watch } from 'vue';
 import { useI18n } from 'vue-i18n';
 
-type Image = {
-	type: string;
-	filesize: number;
-	filename_download: string;
-	width: number;
-	height: number;
-};
+type Image = Pick<
+	File,
+	'type' | 'filesize' | 'filename_download' | 'width' | 'height' | 'focal_point_x' | 'focal_point_y'
+>;
```

Comment:
```suggestion
const imageFields = ['type', 'filesize', 'filename_download', 'width', 'height', 'focal_point_x', 'focal_point_y'] as (keyof File)[];

type Image = Pick<File, (typeof imageFields)[number]>;
```

Reply:
For me this leads to broken types.

1. It doesnt error if you include non-existing keys. So if `File` changes we wont be notified of errors.
2. It doesnt pick your selected keys, it picks all of the `File` keys, hence gives you completion for non-existing keys. We only want to fetch a subset.

Per 1)

![image](https://github.com/directus/directus/assets/14810858/56dec8ff-55c9-40ce-b30c-c3ec74299f5c)

Per 2)

![image](https://github.com/directus/directus/assets/14810858/fd04c017-7320-458b-b7c4-4245fa88b4f5)

Doing it explicitly gives us the correct type and reports back on errors for non-existing keys and will error in the future if `File` changes.

![image](https://github.com/directus/directus/assets/14810858/3c903bd1-8bdc-42a3-a664-c3eb8161a760)

Reply:
did i make a mistake in my type suggestion? 😅 It should be:
```ts
const imageFields = [
	'type',
	'filesize',
	'filename_download',
	'width',
	'height',
	'focal_point_x',
	'focal_point_y',
] as const;

type Image = Pick<
	File,
	typeof imageFields[number]
>;
```

![image](https://github.com/directus/directus/assets/9389634/44fcab72-a747-4660-8cda-1c764ce64081)
![image](https://github.com/directus/directus/assets/9389634/e7ae7d80-214e-463f-bfb6-daa877da65e4)

Reply:
Ahhhh nice! Thanks!